.github/actions/lint-java/action.yml

```diff
@@ -18,15 +18,23 @@ inputs:
   java_version:
     description: 'The version of Java to install and use.'
     type: 'string'
-    required: true
+    default: ''
+  java_version_file:
+    description: 'The version file to use to detect java versions.'
+    type: 'string'
+    default: '.java-version'
   java_distribution:
     description: 'The distibution of Java to use.'
     type: 'string'
-    default: 'zulu'
+    default: ''
   google_java_format_version:
     description: 'The version of google-java-format to use. This must be the full version with no leading "v" prefix.'
     type: 'string'
     default: '1.25.2'
+  google_java_format_version_file:
```

Comment:
I'm not convinced we want to allow customizing this via a file? `.java-version` is an industry standard file. `.java-version-google-format` is not.

Reply:
> Its possible there is a way to resolve this without changing google format versions - I didn't explore. Maybe we can target a specific version of java to match the format version that we want and have it lint for a target java version?

I'd rather do that if posssible